pkg/export/setup/setup.go

```diff
@@ -230,3 +230,73 @@ func loadKubeConfig(kubeconfigPath string) (*rest.Config, error) {
 func ExtraArgs() ([]string, error) {
 	return shlex.Split(os.Getenv(ExtraArgsEnvvar))
 }
+
+// tryPopulateUnspecifiedFromMetadata assumes we are in GCP, with unknown state
+// of the metadata server. This is best-effort, so when the metadata server
+// is not accessible on GCP, because it was disabled (404 errors), or not accessible
+// (connection refused, slow network, e.g. sandbox + metadata disabled)
+// it's a noop. Make sure to pass context with a timeout.
+func tryPopulateUnspecifiedFromMetadata(ctx context.Context, logger log.Logger, opts *export.ExporterOpts) {
```

Comment:
I doubt we hit the same problem, but that would be a good practice and little harm! I want to keep this PR small for bugfix though, let me check if this can be still "small" with operator changes.

Reply:
Ok, I checked and indeed it would be nice, but I would postpone this PR until https://github.com/googleapis/google-cloud-go/pull/10370 is merged (if anything). Will be easier. For now we could merge this and patch 0.12.0?